opensfm/src/geometry/camera.h

```diff
@@ -20,43 +19,37 @@ class Camera {
   Eigen::Vector3d Bearing(const Eigen::Vector2d& point) const;
   Eigen::MatrixX3d BearingsMany(const Eigen::MatrixX2d& points) const;
 
+  void SetProjectionParams(const Eigen::VectorXd& projection);
+  const Eigen::VectorXd& GetProjectionParams() const;
+
+  void SetDistortion(const Eigen::VectorXd& distortion);
+  const Eigen::VectorXd& GetDistortion() const;
+
+  void SetPrincipalPoint(const Eigen::Vector2d& principal_point);
+  Eigen::Vector2d GetPrincipalPoint() const;
+
+  void SetFocal(double focal);
+  double GetFocal() const;
+
+  void SetAspectRatio(double focal);
+  double GetAspectRatio() const;
+
+  ProjectionType GetProjectionType()const;
+  std::string GetProjectionString()const;
+
+  Eigen::Matrix3d GetProjectionMatrix()const;
+  Eigen::Matrix3d GetProjectionMatrixScaled(int width, int height)const;
+  
+  int width{1};
+  int height{1};
+  std::string id;
+
  private:
   Camera();
 
-  // This is where the pseudo-strategy pattern takes place.
-  template <class OUT, class FUNC, class... IN>
-  OUT Dispatch(IN... args) const;
-
-  struct ProjectT {
-    template <class PROJ, class DISTO, class AFF>
-    static Eigen::Vector2d Apply(const Eigen::Vector3d& point,
-                                 const Eigen::VectorXd& projection,
-                                 const Eigen::Matrix2d& affine,
-                                 const Eigen::Vector2d& principal_point,
-                                 const Eigen::VectorXd& distortion) {
-      return AFF::Forward(
-          DISTO::Forward(PROJ::Forward(point, projection), distortion), affine,
-          principal_point);
-    }
-  };
-
-  struct BearingT {
-    template <class PROJ, class DISTO, class AFF>
-    static Eigen::Vector3d Apply(const Eigen::Vector2d& point,
-                                 const Eigen::VectorXd& projection,
-                                 const Eigen::Matrix2d& affine,
-                                 const Eigen::Vector2d& principal_point,
-                                 const Eigen::VectorXd& distortion) {
-      return PROJ::Backward(
-          DISTO::Backward(AFF::Backward(point, affine, principal_point),
-                          distortion),
-          projection);
-    }
-  };
-
-  Type type_;
+  ProjectionType type_;
   Eigen::VectorXd projection_;      // dual transition (1 = perspective, 0 = fisheye)
-  Eigen::Matrix2d affine_;          // fx 0, 0, fy
+  Eigen::Matrix2d affine_;          // fx, skew, skew, fy = (ar*fx)
```

Comment:
should the matrix be upper triangular?
```suggestion
  Eigen::Matrix2d affine_;          // fx, skew, 0, fy = (ar*fx)
```

Reply:
Depends. We can support either the symmetric case or not, or both. I'll add this remark for later.